security/pkg/nodeagent/caClient/client.go

```diff
@@ -37,27 +43,45 @@ type caClient struct {
 }
 
 // NewCAClient create an CA client.
-func NewCAClient(addr string, dialOptions []grpc.DialOption) (Client, error) {
-	conn, err := grpc.Dial(addr, dialOptions...)
+func NewCAClient(endpoint, rootPath string) (Client, error) {
+	var opts grpc.DialOption
+	if rootPath != "" {
+		creds, err := credentials.NewClientTLSFromFile(rootPath, "")
+		if err != nil {
+			log.Errorf("Failed to read root certificate file: %v", err)
+			return nil, fmt.Errorf("failed to read root cert")
+		}
+		opts = grpc.WithTransportCredentials(creds)
+	} else {
```

Comment:
What's the purpose of supporting this option? Ideally we should only provide secure one.

Reply:
Generally we don't add logic in code to facilitate the unit test implementation.  

Here is some example for grpc server with creds and hope it can be a useful reference
https://github.com/istio/istio/blob/master/security/pkg/platform/aws_test.go#L257

Reply:
good catch wattli

Reply:
The PR got merged automatically, please try to address this comment in a follow-up PR when available.